docs/guide/index.md

```diff
@@ -67,8 +67,8 @@ You can also directly specify the project name and the template you want to use
 # npm 6.x
 npm create vite@latest my-vue-app --template vue
 
-# npm 7+, extra double-dash is needed:
-npm create vite@latest my-vue-app -- --template vue
+# npm 7+, an escaping double-dash is needed:
+npm create -- vite@latest my-vue-app --template vue
```

Comment:
I find this confusing in contrary even if it's technically more correct. Usually it's taught as "if you need to pass extra args, precede with -- before the args", which prevents backtracking while you type. Plus the npm docs [recommend the former too](https://docs.npmjs.com/cli/v9/commands/npm-init#forwarding-additional-options) (links to npm init, which is what npm create supersedes). I think we should keep it as is.

Reply:
Every time you run a command line arguments parser it will consume 1 `--`.

`npm <subcommand<` will run the parser once. Vite then also has its own parser which will run _after_ the npm parser.

Surely there's a vite plugin that also requires escaping, and it would need a distinct `--`.

That would also mean that you actually 

```sh
npm create -- vite -- <subcommand> --flag
```

You might not like the aesthetics of where the dash goes, but the purpose isn't to be pretty, it's to communicate intent - which of the nested commands _can_ consume the `--`.

Furthermore, if npm fixes the bug in their parser to not consume `--` after the first unknown argument, then the example as written,m will break.

Ganted, they _probably_ won't because that would require reading `package.json.scripts` and maybe some stuff in `node_modules` to figure out which commands in can load before running the parser... that said, they actually _did_ just change the order of parsing and processing in the version released a few weeks ago which _did_ break some run scripts so there's that.

Better to err 1st on the way things work in practice and 2nd on the side of the standards that have existed since the first unixes in the 1980s than to change things around.

If you find it confusing because learned how CLI parsing works and only by encountering examples that lead to inconsistent and buggy behavior, don't you think it's better to not propagate that confusion for everyone else too?

- There should be one right way to do it
- Readability counts

This is the way.

Reply:
As the npm docs doesn't recommend so, our docs should stick with what we have now. If you think the syntax is incorrect, I'd suggest fixing it in the npm docs first. Right now, most if not all of the npm ecosystem uses the syntax Vite and npm shows.